xml/templates/message_templates/membership_online_receipt_text.tpl

```diff
@@ -154,14 +154,14 @@
 {/foreach}
 {/if}
 
-{if !( $contributeMode eq 'notify' OR $contributeMode eq 'directIPN' ) and $is_monetary}
+{if $email}
 {if $is_pay_later}
 ===========================================================
 {ts}Registered Email{/ts}
 
 ===========================================================
 {$email}
-{elseif $amount GT 0 OR $membership_amount GT 0 }
+{elseif $billingName}
 ===========================================================
```

Comment:
So I left this out to think about it more but as you see it is possible to have email AND billing name. I think we need to reverse it to be 

{if billingName}

{elseif $email}

{/if}

(you might want to merge mine & rebase this & focus in on the small number of trickier ones)

Reply:
@eileenmcnaughton that will still work in mine because note the elseif i believe is linked to the `{if $is_pay_later}` not the `{if $email}`